perma_web/api2/views.py

```diff
@@ -0,0 +1,483 @@
+    class Meta:
+        model = Link
+        fields = ['url', 'date', 'min_date', 'max_date']
+
+
+# /public/archives
+class PublicLinkListView(BaseView):
+    permission_classes = ()  # no login required
+    serializer_class = LinkSerializer
+    filter_class = LinkFilter
+    search_fields = ('guid', 'submitted_url', 'submitted_title', 'notes')  # fields that can be searched with q= query string
+
+    def get(self, request, format=None):
+        """ List public links. """
+        queryset = Link.objects.order_by('-creation_timestamp').prefetch_related('captures').discoverable()
+        return self.simple_list(request, queryset)
+
+# /public/archives/:guid
+class PublicLinkDetailView(BaseView):
+    permission_classes = ()  # no login required
+    serializer_class = LinkSerializer
+
+    def get(self, request, guid, format=None):
+        """ Get public link details. """
+        try:
+            obj = Link.objects.discoverable().get(pk=guid)
+        except Link.DoesNotExist:
+            raise Http404
+        return self.simple_get(request, obj=obj)
+
+# /archives
+# /folders/:parent_id/archives
+class AuthenticatedLinkListView(BaseView):
+    serializer_class = AuthenticatedLinkSerializer
+    filter_class = LinkFilter
+    search_fields = PublicLinkListView.search_fields + ('notes',)  # private links can also be searched by notes field
+
+    @staticmethod
+    def get_folder_from_request(request):
+        """
+            Helper method to load folder from request.data['folder'].
+            Used by AuthenticatedLinkListView.post and AuthenticatedLinkDetailView.patch.
+        """
+        if request.data.get('folder'):
+            try:
+                return Folder.objects.accessible_to(request.user).get(pk=request.data['folder'])
+            except Folder.DoesNotExist:
+                raise ValidationError({'folder': ["Folder not found."]})
+        return None
+
+    @load_parent
+    def get(self, request, format=None):
+        """ List links for user. """
+        queryset = Link.objects\
+            .order_by('-creation_timestamp')\
+            .select_related('organization', 'organization__registrar','capture_job')\
+            .prefetch_related('captures')\
+            .accessible_to(request.user)
+
+        # for /folders/:parent_id/archives, limit to links in folder
+        if request.parent:
+            queryset = queryset.filter(folders=request.parent)
+
+        return self.simple_list(request, queryset)
+
+    @load_parent
+    def post(self, request, format=None):
+        """ Create new link. """
+        data = request.data
+
+        # Set target folder, in order of preference:
+        # - 'folder' key in data
+        # - parent folder, if posting to /folders/:parent_id/archives
+        # - user's personal folder
+        folder = self.get_folder_from_request(request) or request.parent or request.user.root_folder
+
```

Comment:
What would you think of a future feature that, if folder is specified via post data and if posted to /folders/:parent_id/archives but the specified folder is not a child of request.parent, throws a validation error?